entproto/internal/todo/ent/proto/entpb/entpb_user_service.go

```diff
@@ -449,20 +462,126 @@ func (svc *UserService) List(ctx context.Context, req *ListUserRequest) (*ListUs
 				[]byte(fmt.Sprintf("%v", entList[len(entList)-1].ID)))
 			entList = entList[:len(entList)-1]
 		}
-		var pbList []*User
-		for _, entEntity := range entList {
-			pbEntity, err := toProtoUser(entEntity)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "internal error: %s", err)
-			}
-			pbList = append(pbList, pbEntity)
+		protoList, err := toProtosUser(entList)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "internal error: %s", err)
 		}
 		return &ListUserResponse{
-			UserList:      pbList,
+			UserList:      protoList,
 			NextPageToken: nextPageToken,
 		}, nil
 	default:
 		return nil, status.Errorf(codes.Internal, "internal error: %s", err)
 	}
 
 }
+
+// BatchCreate implements UserServiceServer.BatchCreate
+func (svc *UserService) BatchCreate(ctx context.Context, req *BatchCreateUsersRequest) (*BatchCreateUsersResponse, error) {
+	requests := req.GetRequests()
+	if len(requests) > entproto.MaxBatchCreateSize {
+		return nil, status.Errorf(codes.InvalidArgument, "batch size cannot be greater than %d", entproto.MaxBatchCreateSize)
+	}
+	bulk := make([]*ent.UserCreate, len(requests))
+	for i, req := range requests {
+		user := req.GetUser()
```

Comment:
can this generated code be extracted to a generated function that is used by both Create and BatchCreate?